### DISCUSSION TITLE
extend usage of KeeLoq (Jarolift shutter working with keeloq #7788) to Sonoff RF Bridge 433

### DISCUSSION THREAD

**Discussion by iot-sle** [arendst/Tasmota#10973] [Ideas]
With Jarolift shutter working with keeloq #7788 it's possible to handle shutters with keeloq mechanism. Unfortunately it is linked to the CC1101 hardware. I did some test with RFRaw commands on a Sonoff RF Bridge 433 flashed with Tasmota and Portisch firmware for the transceiver part. And yes prinicipally it CAN work.  In details sometimes I can control the shutters. But after a short time the control is stopped. (As soon as I'm using the remotes for controling in between). The reason is the rolling code mechanism based on keeloq. Just that part was implemented in #7788  but only linked to the CC1101 transmitter hardware. So it should be possible to use the exiting code and extend the "sending" / "receiving" possibilites by using the coding routines together with RFRaw sending and receiving. In xrdv_36_keeloq.ino the coding and decoding of keeloq is already separated. Perhaps somebody does know how to generate the right RFRaw in put from output of CreateKeeloqPacket() function?

**Comment by Bascht74** [comment]
@iot-sle I have the same problem and right now a beginner with Jarolift shutters.

Can't you use a different serial number for the RF Bridge and learn the shutters to that serial so that rolling codes of the remote doesn't matter anymore?
Or use a remote that you don't use afterwards.
Is it the case, that the rolling codes do matter if you use different remotes?
Maybe you found a trick since February :-)

Hallo iot-sle, ich habe das gleiche Problem und aktuell bin ich noch ein Anfänger und habe eine Jarolift Markise.
Kannst Du nicht eine andere Seriennummer verwenden und die RF Bridge einlernen, so dass die rolling codes der anderen Fernbedienungen keine Rolle spielen? Oder könnte man eine andere Fernbedienung einlernen und dessen Serial verwenden und diese Fernbedienung danach nicht mehr benutzen?
Oder ist es so, dass die rolling codes auch eine Rolle spielen, wenn man andere Fernbedienungen benutzt!
Evtl. hast du das ganze ja inzwischen irgendwie gelöst oder einen Trick gefunden... :-)

**Comment by iot-sle** [comment]
The correct way will be to use a different serial number for the RF-Bridge and add the RF-Bridge as a new remote to each Jarolift shutter. As you would do using  several remotes.

Unfortunately the rolling code is not implemented in the Sonoff RF Bridge until now. I wanted to do that during the last month, but I was short on time. Maybe I will be able to do that end of summer.

 

 

In Deutsch: Die Jarolifter Rollos sind mit dem rolling code abgesichert. Daher muss jede Fernbedienung an dem zugehörigen Rollo registriert werden, also auch die RF-Bridge. Jede Bedienung bekommt, dann ihre eigenen rolling code. Ich hatte am Anfang das Gefühl, die Jarolifter Rollos würden nur 3 oder 4 verschiedene Codes benutzen. Das würde ein „schmutzige“ Implementierung

Also Kopie einer vorhanden Fernbedienung erlauben, aber schön ist das nicht. Daher wäre das richtige das rolling code Verfahren aus dem keeloq modul in das RF-Bridge Modul zu übernehmen und dann die RF-Bridge als weitere Fernbedienung mit eigenem rolling code in den Rollos zu registrieren. Leider hat meine Zeit es nicht zugelassen. Den rolling code Mechanismus aus dem keeloq modul in das RF-Bridge modul zu übertragen. Das wir auch frühestens am Ende des Sommers etwas. 

 

Viele Grüße

 

Von: Bascht74 ***@***.*** 
Gesendet: Dienstag, 22. Juni 2021 12:18
An: arendst/Tasmota
Cc: iot-sle; Mention
Betreff: Re: [arendst/Tasmota] extend usage of KeeLoq (Jarolift shutter working with keeloq #7788) to Sonoff RF Bridge 433 (#10973)

 

@iot-sle <https://github.com/iot-sle>  I have the same problem and right now a beginner with Jarolift shutters.

Can't you use a different serial number for the RF Bridge and learn the shutters to that serial so that rolling codes of the remote doesn't matter anymore?
Or use a remote that you don't use afterwards.
Is it the case, that the rolling codes do matter if you use different remotes?
Maybe you found a trick since February :-)

Hallo iot-sle, ich habe das gleiche Problem und aktuell bin ich noch ein Anfänger und habe eine Jarolift Markise.
Kannst Du nicht eine andere Seriennummer verwenden und die RF Bridge einlernen, so dass die rolling codes der anderen Fernbedienungen keine Rolle spielen? Oder könnte man eine andere Fernbedienung einlernen und dessen Serial verwenden und diese Fernbedienung danach nicht mehr benutzen?
Oder ist es so, dass die rolling codes auch eine Rolle spielen, wenn man andere Fernbedienungen benutzt!
Evtl. hast du das ganze ja inzwischen irgendwie gelöst oder einen Trick gefunden... :-)

**Reply by Bascht74** [reply]
@iot-sle 
Hi, I have worked on it a little in the meantime. 
I found out the MasterMSB and MasterLSB and stored them together with a Serial and the Counter via the command `keeloqset 65xxxxx27 29xxxxx37 160xxx32 0`.

I had extended the xdrv_36_keeloq.ino at first to give out the B0 code for the RF bridge. 

Is there any way we can exchange information privately (e.g. discord)? 
I'd love to hear your thoughts on this.

Cheers,

Sebastian

in Deutsch:
Hi, ich habe inzwischen ein wenig weiter gearbeitet. Ich habe den MasterLSB und MasterMSB herausbekommen und zusammen mit einer Serial und dem Counter über den Befehl `keeloqset 65xxxxx27 29xxxxx37 16034032 0` eingelernt. 
Dazu habe ich die xdrv_36_keeloq.ino erweitert, dass sie zunächst einmal den B0-Code für die RF-Bridge herausgibt. 
Können wir uns irgendwie privat austauschen? Ich würde gerne Deine Meinung dazu hören.

Liebe Grüße

Sebastian

**Reply by Bascht74** [reply]
@iot-sle 
Now I am able to "copy" an existing remote (found out serial and count) and generate the B0 code that is exactly the code the bridge picks up as B1 with rf 177.
Now I will look into calling the right stuff to send the B0 code directly to the transmitter. 

Btw.: 
I was not able to learn a new serial (with button 1) so far. I will look into that. I don't know why it is not working.
Already learned remote is fine now. I can now call all the buttons I want and use the shutter component in Tasmota...

Seb

**Reply by Bascht74** [reply]
Calling the transmitter is working now. 
Next step is to analyse the learn code of an existing remote to see if it is the serial number that is causing the problems or the B0 code itself...
Code is cleaned up a little bit ( I am no programmer)

**Reply by Bascht74** [reply]
here is what I got so far: 
https://github.com/Bascht74/Tasmota/blob/development/tasmota/xdrv_36_keeloq.ino

**Reply by hugoheinzson** [reply]
guys, is there anyone to find out the MasterLSB and other Codes I need? I'm a coding noob just trying to figure out to controll my jarolist markise with a SignalDuino from my Raspberry Pi with Home Assistant...

**Reply by Bascht74** [reply]
> guys, is there anyone to find out the MasterLSB and other Codes I need? I'm a coding noob just trying to figure out to controll my jarolist markise with a SignalDuino from my Raspberry Pi with Home Assistant...

Search for AE77FCACBED3BEEC14933EB7FF1F942F (this ist not the code) and look at the results in google.

They look like this in hex-format: 

  MSB: '0x2...B'
  LSB: '0x1...5'

You can convert them to decimal if you need...

This could be an alternative project if you use home assistant: https://github.com/dewenni/ESP32-Jarolift-Controller

**Comment by Bascht74** [comment]
@iot-sle anything new?

**Comment by Stevelande** [comment]
https://github.com/madmartin/Jarolift_MQTT
There was already a version here. In this version, the CC1101 was taught as an extra remote control. Maybe you can find the necessary code lines or whatever here.
I have rebuilt it. But everything was not stable. Now I hoped that I find a solution with Tasmota and Portisch.
Would also communicate privately!

**Reply by Stevelande** [reply]
@Bascht74 find you a solution? Can we maybe have contact?

**Reply by Bascht74** [reply]
> here is what I got so far: https://github.com/Bascht74/Tasmota/blob/development/tasmota/xdrv_36_keeloq.ino

As I said, I am no programmer. I used the CC1101 Tasmota-Code and generated the B0-Code for a portish flashed Sonoff RF Bridge 433. It just shows how it could work if somebody with advanced programming skills could invest 2-3 hours...

**Reply by Stevelande** [reply]
I don't think anyone will be able to find one.
Too bad. The Jarolift mqtt thing does not run too reliably for me. A solution would still be cUL together with FHEM .
